aws/kops-legacy-account-vpc-peering/main.tf

```diff
@@ -0,0 +1,40 @@
+terraform {
+  required_version = ">= 0.11.2"
+
+  backend "s3" {}
+}
+
+provider "aws" {
+  assume_role {
+    role_arn = "${var.aws_assume_role_arn}"
+  }
+}
+
+# Lookup VPC of the kops cluster
+module "kops_metadata" {
+  source   = "git::https://github.com/cloudposse/terraform-aws-kops-metadata.git?ref=tags/0.2.1"
+  dns_zone = "${var.region}.${var.zone_name}"
+  enabled  = "${var.enabled}"
+}
+
+module "kops_legacy_account_vpc_peering" {
+  source = "git::https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account.git?ref=tags/0.5.0"
+
+  enabled = "${var.enabled}"
+
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "${var.name}"
+
+  auto_accept = true
+
+  # Requester
+  requester_vpc_id              = "${module.kops_metadata.vpc_id}"
```

Comment:
I don't think this actually works. It looks to me like `terraform-aws-kops-metadata` is going to return all the VPCs and you are just getting lucky that there is only one to return.

Reply:
We use it everywhere.

https://github.com/cloudposse/terraform-root-modules/blob/master/aws/backing-services/elasticsearch.tf#L58
https://github.com/cloudposse/terraform-root-modules/blob/master/aws/backing-services/aurora-postgres.tf#L92

It returns only one VPC.

If we are changing the way we name and tag `kops` resources, we need to discuss that since it could introduce breaking changes.

Reply:
`kops`/`kubernetes` 1.12 does not tag the VPC when it does not create the VPC and provides no consistent way to determine the VPC it is using. Probably the right place to address this in the long run is `terraform-aws-kops-metadata` but for now let's at least pass through [`vpc_tag` and `vpc_tag_vaules`](https://github.com/cloudposse/terraform-aws-kops-metadata/blob/cea5cba634b2659aefb6a2e6821161d7081baa1d/variables.tf#L29-L37)